src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,817 @@
+        elif isinstance(x, Trial):
+            x = x.params
+        elif isinstance(x, tuple) and len(x) == len(self._space):
+            x = dict(zip(self._space.keys(), x))
+        elif isinstance(x, dict):
+            # NOTE: Passing ndarrays and tensors is useful for testing, but this task now probably
+            # accepts too many types of inputs.
+            x = x
+        elif isinstance(x, (np.ndarray, Tensor)):
+            if x not in self._space:
+                warnings.warn(
+                    RuntimeWarning(
+                        f"Point {x} isn't a valid point of the space {self._space}, but will still "
+                        f"be passed to the task's model."
+                    )
+                )
+        else:
+            raise ValueError(
+                f"Expected `x` to be a dataclass, a Trial, a dict, or a tuple of length "
+                f"{len(self._space)}, but got {x} instead."
+            )
+
+        if isinstance(x, dict):
+            # A bit of gymnastics to convert the params Dict into a numpy array.
+            trial = dict_to_trial(x, self._space)
+            point_tuple = trial_to_tuple(trial, self._space)
+            flattened_point = flatten_dims(point_tuple, self._space)
+            x = np.array(flattened_point)
+
+        if isinstance(x, np.ndarray):
+            x = torch.as_tensor(x, dtype=torch.float32, device=self.device)
+
+        assert isinstance(
+            x, Tensor
+        )  # Just to keep the type checker happy: this is always true.
+        x_tensor: Tensor = x
+        if with_grad:
+            x_tensor.requires_grad_(True)
+
+        x_tensor = x_tensor.type_as(self.h_tensor)
+        p_tensor = torch.cat([x_tensor, self.h_tensor])
+        p_tensor = torch.atleast_2d(p_tensor)
+
+        with self.seed_randomness():
+            # TODO: Bug when using forrester task:
+            # RuntimeError: size mismatch, m1: [1 x 4], m2: [3 x 100]
+            out = self.net(p_tensor)
+            # TODO: Double-check that the std is indeed in log form.
+            y_mean, y_log_std = out[0, 0], out[0, 1]
+            y_std = torch.exp(y_log_std)
+
+            # NOTE: Here we create a distribution over `y`, and use `rsample()`, so that we get can also
+            # return the gradients if need be.
+            y_dist = Normal(loc=y_mean, scale=y_std)
+
+            y_sample = y_dist.rsample()
+            logger.debug(f"y_sample: {y_sample}")
+
+        results: List[Dict] = []
+
+        results.append(dict(name=self.name, type="objective", value=float(y_sample)))
+
+        if with_grad:
+            self.net.zero_grad()
+            y_sample.backward()
+            results.append(
+                dict(name=self.name, type="gradient", value=x_tensor.grad.cpu().numpy())
+            )
+
+        return results
+
+    @property
+    def configuration(self):
+        """Return the configuration of the task."""
+        # TODO: Still not sure I understand
```

Comment:
Can I help? :)

Reply:
Fixed. I still find it a bit weird to have to use this Factory class for the serialization/deserialization though.

Reply:
It helps with the creation of objects and support for backward compatible configurations in database. Supporting backward compatibility with pickle-like serialization can be quite painful. :sweat_smile: